src/components/utils/ImgProfile/ImgProfile.tsx

```diff
@@ -1,19 +1,19 @@
 import React, { useCallback, useEffect, useState } from 'react';
-import { CV, UserWithUserCandidate } from 'src/api/types';
+import { CV, PublicProfile, UserWithUserCandidate } from 'src/api/types';
+import { Img } from 'src/components/utils/Img';
 import { CANDIDATE_USER_ROLES } from 'src/constants/users';
 import { useAuthenticatedUser } from 'src/hooks/authentication/useAuthenticatedUser';
 import { isRoleIncluded } from 'src/utils';
-import { Img } from './Img';
 
 interface ImgProfileProps {
-  user: UserWithUserCandidate;
+  user: UserWithUserCandidate | PublicProfile;
```

Comment:
Est-ce que ici, on ne changerai pas plutôt le type de en entrée, en mettant `Partial<UserWithUserCandidate>`. Comme ça, tu peux passer que les props sur le user que tu as dans le profile et qui t'intéresse. Ça évite d'essayer de gérer 2 types qui sont complètement différents `UserWithUserCandidate | PublicProfile`. Et comme ça pas besoin de checker le `'candidat' in candidatUser`

Reply:
Ca me pose pas de souci en soi de faire comme tu procèdes, mais si tu passes les props sur le user qui t'intéressent dans le profile, tu n'auras pas l'objet "candidat" donc tu seras obligé de le vérifier, non?

Reply:
Du coup la props `candidatUser` sera optionnel, donc il suffira juste de vérifier si elle est défini `if(candidatUser) ...`

Reply:
Ben c'est plus compliqué que ça, parce que du coup tu rends tous les champs de "UserWithUserCandidate" optionnels ce qui demande de vérifier qu'ils ne sont pas indéfinis à chaque utilisation (comme `isRoleIncluded(CANDIDATE_USER_ROLES, candidatUser.role)`).
Et ca n'empêche qu'il faudra continuer à vérifier si "candidat" existe dans "candidatUser", puisque candidat reste un champ optionnel de "Partial<UserWithUserCandidate>".

Reply:
Oui effectivement mais du coup tu check pas en mode `'candidat' in candidatUser`, mais tu check en mode `if (candidatUser?.candidat?.role) ou if(candidatUser?.candidat?.cvs)`  (comme on fait déjà)